java/vector/src/test/java/org/apache/arrow/vector/complex/writer/TestComplexWriter.java

```diff
@@ -561,4 +564,76 @@ public void mapWriterMixedCaseFieldNames() {
+
+    TimeStampWriter timeStampWriter = rootWriter.timeStamp("millis");
+    timeStampWriter.setPosition(1);
+    timeStampWriter.writeTimeStamp(1L);
+
+    TimeStampMicroWriter timeStampMicroWriter = rootWriter.timeStampMicro("micro");
+    timeStampMicroWriter.setPosition(2);
+    timeStampMicroWriter.writeTimeStampMicro(2000L);
+
+    TimeStampNanoWriter timeStampNanoWriter = rootWriter.timeStampNano("nano");
+    timeStampNanoWriter.setPosition(3);
+    timeStampNanoWriter.writeTimeStampNano(3000000L);
+
+    // schema
+    Field secField = parent.getField().getChildren().get(0).getChildren().get(0);
+    Assert.assertEquals("sec", secField.getName());
+    Assert.assertEquals(ArrowType.Timestamp.TYPE_TYPE, secField.getType().getTypeID());
+
+    Field millisField = parent.getField().getChildren().get(0).getChildren().get(1);
+    Assert.assertEquals("millis", millisField.getName());
+    Assert.assertEquals(ArrowType.Timestamp.TYPE_TYPE, millisField.getType().getTypeID());
+
+    Field microField = parent.getField().getChildren().get(0).getChildren().get(2);
+    Assert.assertEquals("micro", microField.getName());
+    Assert.assertEquals(ArrowType.Timestamp.TYPE_TYPE, microField.getType().getTypeID());
+
+    Field nanoField = parent.getField().getChildren().get(0).getChildren().get(3);
+    Assert.assertEquals("nano", nanoField.getName());
+    Assert.assertEquals(ArrowType.Timestamp.TYPE_TYPE, nanoField.getType().getTypeID());
+
+    // read
+    MapReader rootReader = new SingleMapReaderImpl(parent).reader("root");
+
+    FieldReader secReader = rootReader.reader("sec");
+    secReader.setPosition(0);
+    DateTime secDateTime = secReader.readDateTime();
+    Assert.assertEquals(new DateTime(0L, DateTimeZone.UTC).withZoneRetainFields(DateTimeZone.getDefault()), secDateTime);
+    long secLong = secReader.readLong();
+    Assert.assertEquals(0L, secLong);
+
+    FieldReader millisReader = rootReader.reader("millis");
+    millisReader.setPosition(1);
+    DateTime millisDateTime = millisReader.readDateTime();
+    Assert.assertEquals(new DateTime(1L, DateTimeZone.UTC).withZoneRetainFields(DateTimeZone.getDefault()), millisDateTime);
+    long millisLong = millisReader.readLong();
+    Assert.assertEquals(1L, millisLong);
+
+    FieldReader microReader = rootReader.reader("micro");
+    microReader.setPosition(2);
+    DateTime microDateTime = microReader.readDateTime();
+    Assert.assertEquals(new DateTime(2L, DateTimeZone.UTC).withZoneRetainFields(DateTimeZone.getDefault()), microDateTime);
+    long microLong = microReader.readLong();
+    Assert.assertEquals(2000L, microLong);
+
+    FieldReader nanoReader = rootReader.reader("nano");
+    nanoReader.setPosition(3);
+    DateTime nanoDateTime = nanoReader.readDateTime();
+    Assert.assertEquals(new DateTime(3L, DateTimeZone.UTC).withZoneRetainFields(DateTimeZone.getDefault()), nanoDateTime);
+    long nanoLong = nanoReader.readLong();
+    Assert.assertEquals(3000000L, nanoLong);
```

Comment:
I am not very sure if I understand the question correctly. I thought accessor.get(index) can be used to read long values while accessor.getObject(index) is used to read DateTime objects.